src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -213,94 +220,74 @@ def resolve_dependencies(self, main_ctx: AnalyzeContext) -> dict[str, Dependency
         dict[str, DependencyInfo]
             A dictionary where artifacts are grouped based on ``artifactId:groupId``.
         """
-        deps_resolved = {}
+        deps_resolved: dict[str, DependencyInfo] = {}
 
-        if "dependency.resolver" not in defaults or "dep_tool_maven" not in defaults["dependency.resolver"]:
-            logger.info("No default dependency analyzer is found. Skipping automatic dependency analysis...")
-        elif not DependencyAnalyzer.tool_valid(
-            defaults.get("dependency.resolver", "dep_tool_maven", fallback="cyclonedx-maven:2.6.2")
-        ):
+        build_tool = main_ctx.dynamic_data["build_spec"]["tool"]
+        if not build_tool or isinstance(build_tool, NoneBuildTool):
+            logger.info("Unable to find a valid build tool.")
+            return {}
+
+        try:
+            dep_analyzer = build_tool.get_dep_analyzer(main_ctx.repo_path)
+        except DependencyAnalyzerError as error:
+            logger.error("Unable to find a dependency analyzer: %s", error)
+            return {}
+
+        if isinstance(dep_analyzer, NoneDependencyAnalyzer):
             logger.info(
-                "Dependency analyzer %s is not valid. Skipping automatic dependency analysis...",
-                defaults.get(
-                    "dependency.resolver",
-                    "dep_tool_maven",
-                    fallback="cyclonedx-maven:2.6.2",
-                ),
+                "Dependency analyzer is not available for %s",
+                main_ctx.dynamic_data["build_spec"]["tool"].name,
             )
-        else:
+            return {}
+
+        # Start resolving dependencies.
+        logger.info(
+            "Running %s version %s dependency analyzer on %s",
+            dep_analyzer.tool_name,
+            dep_analyzer.tool_version,
+            main_ctx.repo_path,
+        )
+
+        log_path = os.path.join(
+            global_config.build_log_path,
+            f"{main_ctx.repo_name}.{dep_analyzer.tool_name}.log",
+        )
+
+        # Clean up existing SBOM files.
+        dep_analyzer.remove_sboms(main_ctx.repo_path)
+
+        commands = dep_analyzer.get_cmd()
+        working_dirs: Iterable[Path] = build_tool.get_build_dirs(main_ctx.repo_path)
+        for working_dir in working_dirs:
+            # Get the absolute path to use as the working dir in the subprocess.
+            working_dir = Path(main_ctx.repo_path).joinpath(working_dir)
```

Comment:
Would it be possible if we return the absolute `workfing_dir` directly from `build_tool.get_build_dirs` so that we don't need to recompute the absolute path here (unless the result of `build_tool.get_build_dirs` is also used elsewhere),

Reply:
The reason I'm using relative paths here is for testing. We now use `syrupy` to take a snapshot of the returned paths from this function, and capturing absolute paths would add paths on my own local machines, which would not work.